pkg/k8sutil/k8sutil.go

```diff
@@ -173,6 +173,8 @@ func UpdateStatefulSet(ctx context.Context, sstClient clientappsv1.StatefulSetIn
 	}
 
 	mergeMetadata(&sset.ObjectMeta, existingSset.ObjectMeta)
+	// Propagate annotations set by kubectl. e.g performing a rolling restart.
+	mergeKubectlAnnotations(&sset.Spec.Template.ObjectMeta, existingSset.Spec.Template.ObjectMeta)
```

Comment:
Can we use/improve `mergeMetadata(&sset.Spec.Template.ObjectMeta, existingSset.Spec.Template.ObjectMeta)` for this? I don't see why we need another function doing the same thing.

Also, why do we care only about kubectl annotations and not all annotations? AFAIK annotations and labels should be preserved during STS recreation.

Reply:
Note that the initial `mergeMetadata` operates on the statefulset's metadata. The one added operates on the pod template.

If we propagate all the labels and annotations, then we can change the pod templates manually bypassing the operator.